apps/arweave/src/ar_tx_emitter_worker.erl

```diff
@@ -35,27 +35,11 @@ handle_cast({emit, TXID, Peer, ReplyTo}, State) ->
 			ok;
 		TX ->
 			StartedAt = erlang:timestamp(),
-			TrustedPeers = ar_peers:get_trusted_peers(),
-			PropagatedTX = tx_to_propagated_tx(TX, Peer, TrustedPeers),
-			Release = ar_peers:get_peer_release(Peer),
-			SendFun =
-				case Release >= 42 of
-					true ->
-						fun() ->
```

Comment:
Any idea why the old code wrapped this in a `fun`? Your refactored version is much cleaner - but I can't for the life of me figure out why the old version added the extra `fun` indirection. Can you think of any reason? Perhaps just "historical" - i.e. maybe there used to be a reason, but it was refactored out or otherwise changed at some point in the past and the reason disappeared.

Reply:
yep agree!